common/fixation_entry.go

```diff
@@ -262,82 +267,61 @@ func (fs *FixationStore) getUnmarshaledEntryForBlock(ctx sdk.Context, safeIndex
 }
 
-// FindEntry returns the entry with index and block without changing the refcount
-func (fs *FixationStore) FindEntry(ctx sdk.Context, index string, block uint64, entryData codec.ProtoMarshaler) (error, bool) {
+// FindEntry returns the entry by index and block without changing the refcount
+func (fs *FixationStore) FindEntry(ctx sdk.Context, index string, block uint64, entryData codec.ProtoMarshaler) bool {
 	safeIndex, err := sanitizeIndex(index)
 	if err != nil {
 		details := map[string]string{"index": index}
-		return utils.LavaError(ctx, ctx.Logger(), "FindEntry_invalid_index", details, "invalid non-ascii entry"), false
+		utils.LavaError(ctx, ctx.Logger(), "FindEntry_invalid_index", details, "invalid non-ascii entry")
+		return false
 	}
 
-	// get the unmarshaled entry for block
 	entry, found := fs.getUnmarshaledEntryForBlock(ctx, safeIndex, block)
 	if !found {
-		return types.ErrEntryNotFound, false
+		return false
 	}
 
-	// unmarshal the entry's data
-	err = fs.cdc.Unmarshal(entry.GetData(), entryData)
-	if err != nil {
-		return utils.LavaError(ctx, ctx.Logger(), "FindEntry_cant_unmarshal", map[string]string{"err": err.Error()}, "can't unmarshal entry data"), false
-	}
-
-	return nil, true
+	fs.cdc.MustUnmarshal(entry.GetData(), entryData)
+	return true
 }
 
-// GetEntry returns the latest entry with index and increments the refcount
-func (fs *FixationStore) GetEntry(ctx sdk.Context, index string, entryData codec.ProtoMarshaler) (error, bool) {
+// GetEntry returns the latest entry by index and increments the refcount
+func (fs *FixationStore) GetEntry(ctx sdk.Context, index string, entryData codec.ProtoMarshaler) bool {
 	safeIndex, err := sanitizeIndex(index)
 	if err != nil {
 		details := map[string]string{"index": index}
-		return utils.LavaError(ctx, ctx.Logger(), "GetEntry_invalid_index", details, "invalid non-ascii entry"), false
+		utils.LavaError(ctx, ctx.Logger(), "GetEntry_invalid_index", details, "invalid non-ascii entry")
+		return false
 	}
 
 	block := uint64(ctx.BlockHeight())
 
-	// get the unmarshaled entry for block
 	entry, found := fs.getUnmarshaledEntryForBlock(ctx, safeIndex, block)
 	if !found {
-		return types.ErrEntryNotFound, false
+		return false
 	}
 
-	// unmarshal the entry's data
-	err = fs.cdc.Unmarshal(entry.GetData(), entryData)
-	if err != nil {
-		return utils.LavaError(ctx, ctx.Logger(), "GetEntry_cant_unmarshal", map[string]string{"err": err.Error()}, "can't unmarshal entry data"), false
-	}
+	fs.cdc.MustUnmarshal(entry.GetData(), entryData)
 
 	entry.Refcount += 1
 	fs.setEntry(ctx, entry)
-
-	return nil, true
+	return true
 }
 
-// get entry with index and block with ref decrease
-func (fs *FixationStore) PutEntry(ctx sdk.Context, index string, block uint64, entryData codec.ProtoMarshaler) (error, bool) {
+// PutEntry finds the entry by index and block and decrements the refcount
+func (fs *FixationStore) PutEntry(ctx sdk.Context, index string, block uint64) {
```

Comment:
Not sure I follow on why the user must have the entry before using PutEntry. Isn't there a scenario where the user would want to retrieve the required entry and decrease the refCount automatically?
Also, if we keep this change (that PutEntry doesn't return anything), I think the function should be renamed, because PutEntry is not very indicative on what it's doing.

Reply:
"Put" must always match a previous "Get", therefore:
- it is reasonable to require that the holder (of the reference) knows which entry version to put. 
- it needs only the entry version, not the actual entry - because it is assumed the entry is there (held by holder).
- it must not fail because the entry must have been "Get"ed before (entry not found is a bad bug).
The name "Put" is a common convention as the opposite of "Get": we "Get()" a reference and then we "Put()" it back when done.